src/server/server.py

```diff
@@ -99,7 +113,118 @@
             print('Option reading error (ssl).')
         exit(1)
 
+# Cluster mode is used for revocation
+try:
+    SERVER_OPTS['cluster'] = CONFIG.get('main', 'cluster').split(',')
+except NoOptionError:
+    # Standalone mode
+    PROTO = 'http'
+    if SERVER_OPTS['ssl']:
+        PROTO = 'https'
+    SERVER_OPTS['cluster'] = ['%s://localhost:%s' % (PROTO, SERVER_OPTS['port'])]
+
+try:
+    SERVER_OPTS['clustersecret'] = CONFIG.get('main', 'clustersecret')
+except NoOptionError:
+    # Standalone mode
+    SERVER_OPTS['clustersecret'] = randomString(32)
+
+
 # INDEPENDANT FUNCTIONS
+def get(url):
+    """
+    Rebuilt GET function for CASSH purpose
+    """
+    session = Session()
+    try:
+        req = session.get(url,
+                          headers=REQ_HEADERS,
+                          timeout=REQ_TIMEOUT,
+                          stream=True)
+    except ConnectionError:
+        print('Connection error : %s' % url)
+        req = None
+    return req
+
+def post(url, payload):
+    """
+    Rebuilt POST function for CASSH purpose
+    """
+    session = Session()
+    try:
+        req = session.post(url,
+                           data=payload,
+                           headers=REQ_HEADERS,
+                           timeout=REQ_TIMEOUT)
+    except ConnectionError:
+        print('Connection error : %s' % url)
+        req = None
+    return req
+
+def randomString(stringLength=10):
+    """Generate a random string of fixed length """
+    letters = ascii_lowercase
+    return ''.join(choice(letters) for i in range(stringLength))
+
+def cluster_alived():
+    """
+    This function returns a subset of pingeable node
+    """
+    alive_nodes = list()
+    dead_nodes = list()
+    for node in SERVER_OPTS['cluster']:
+        req = get("%s/ping" % node)
+        if req is not None and req.text == 'pong':
+            alive_nodes.append(node)
+        else:
+            dead_nodes.append(node)
+    return alive_nodes, dead_nodes
```

Comment:
Shouldn't you only return the 'alive' servers rather than a tuple ? The function name is confusing compared to the returned value.

Reply:
`cluster_status()` would be ok ?